test/testdata/rbi/kernel.rb

```diff
@@ -65,9 +65,16 @@ def raise_test
 obj = T.let("foo", String)
 T.assert_type!(obj.itself, String)
 
-y = loop do
-end
-puts y # error: This code is unreachable
+
+# These types are deliberately wrong, because `Kernel#p` is difficult to type
+# in an RBI.  See the comments in kernel.rbi.
```

Comment:
Nit: could you add a comment with the expected type for the integer and string case like the array case (just so it's obvious what `p` is actually returning)?